doc/decisions/DR017-Entity-version-queries.md

```diff
@@ -0,0 +1,213 @@
+
+If a tool wishes to indicate to a user that a new version is available
+for a specific entity, with UI feedback.
+
+| Presentation     | Existing                    | Decomposed                        | Diff |
+| ---------------- | --------------------------- | --------------------------------- | ---- |
+| Existence only   | `entityVersions`            | `getWithRelationship`             | 0    |
+| Version          | `entityVersions`            | `getWithRelationship` + `resolve` | +1   |
+| Version and Date | `entityVersion` + `resolve` | `getWithRelationship` + `resolve` | 0    |
+
+#### Asset export
+
+Traversing the history of an asset, for use in some other detached
+scenario, that includes all trait data.
+
+| Existing                     | Decomposed                        | Difference |
+| ---------------------------- | --------------------------------- | ---------- |
+| `entityVersions` + `resolve` | `getWithRelationship` + `resolve` | 0          |
+
+#### Summary
+
+| Scenario                  | Existing | Decomposed | Cost | Call Volume |
+| ------------------------- | -------- | ---------- | ---- | ----------- |
+| Basic Picker              | 1        | 2          | 1    | Low         |
+| Extended Picker           | 2        | 2          | 0    | Low         |
+| Clip Annotation           | 1        | 1          | 0    | Med         |
+| Clip Versioned Annotation | 2        | 1          | -1   | Med         |
+| Export                    | 2        | 2          | 0    | Low         |
+| Updated check             | 1        | 1          | 0    | Low         |
+| Updated check w/version   | 1        | 2          | 1    | Low         |
+| Updated check w/info      | 2        | 2          | 0    | Low         |
+
+## Options considered
+
+### Option 1 - Keep first-class versions API
+
+With this option, OpenAssetIO would maintain first-class methods to
+introspect entity versions.
+
+#### Pros
+
+- Optimal in situations where the version name is the only required
+  information to be presented to the suer.
+- Less abstract - defining versions as a first-class API concept.
+
+#### Cons
+
+- Formalizes the definition of a version in the core API, extensions to
+  version information (eg: additional metadata) requires a core update.
+- Increases API requests when information other than the name is also
+  required.
+- Larger API surface area to maintain.
+
+### Option 2 - Decompose to basic API methods
+
+With this option, all versioning information would be queried through
+the generic `resolve`, `getWithRelationship` API methods.
+
+#### Pros
+
+- Allows arbitrary data to be queried along-side versioning information,
+  facilitating runtime adaptation of user presentation with constant
+  call count.
+- Allows domain specific definitions of a "versions" without needing a
+  core API change.
+- Smaller API surface area to maintain.
+
+#### Cons
+
+- More abstract.
+- Increases API requests when only the version name is required.
+
```

Comment:
[consider] One more `switch` case in the manager's implementation. But then won't most implementation details be shared between version queries and more generic relationship queries anyway?